background/index.ts

```diff
@@ -163,7 +166,84 @@ async function main() {
     setTimeout(fetchUserOpsSent, timeout)
   }
 
+  // TODO: In the future, adding an Indexer to the Background Script to
+  // monitor Account-related transactions. Updates like balance will trigger
+  // as needed, avoiding fixed interval polling with setInterval().
+  const fetchTokenBalance = async () => {
+    const timeout = 3000
+    console.log(`[background] fetch token balance every ${timeout} ms`)
+
+    const { account, network } = storage.get()
+    const { node } = networkManager.getActive()
+
+    const accountId = network[state.networkActive].accountActive
+    const accountAddress = account[accountId].address
+    const provider = new JsonRpcProvider(node.url)
+    const { tokens } = account[accountId]
+
+    // Update the balance of all tokens
+    for (const t of tokens) {
+      try {
+        const balance = await getErc20Contract(t.address, provider).balanceOf(
+          accountAddress
+        )
+        const balanceHex = number.toHex(balance)
+
+        if (t.balance !== balanceHex) {
+          storage.set((state) => {
+            const token = state.account[accountId].tokens.find((token) =>
+              address.isEqual(token.address, t.address)
+            )
+            if (!token) {
+              console.warn(`[Popup][tokens] Unknown token: ${t.address}`)
+              return
+            }
+            token.balance = balanceHex
+          })
+        }
+      } catch (error) {
+        console.warn(
+          `[Popup][tokens] error occurred while getting balance: ${error}`
+        )
+        continue
+      }
+    }
+
+    setTimeout(fetchTokenBalance, timeout)
+  }
+
+  // TODO: In the future, adding an Indexer to the Background Script to
+  // monitor Account-related transactions. Updates like balance will trigger
+  // as needed, avoiding fixed interval polling with setInterval().
+  const fetchNativeBalance = async () => {
```

Comment:
Maybe it can be combined with `fetchTokenBalance` to `fetchBalance`? There are some duplicate setups in them.